donkeycar/parts/controller.py

```diff
@@ -227,107 +228,138 @@ def poll(self):
     def set_deadzone(self, val):
         self.dead_zone = val
 
-
 class RCReceiver:
-    """
-    Class to read PWM from an RC control and convert into a float output number.
-    Uses pigpio library. The code is essentially a copy of
-    http://abyz.me.uk/rpi/pigpio/code/read_PWM_py.zip. You will need a voltage
-    divider from a 5V RC receiver to a 3.3V Pi input pin if the receiver runs
-    on 5V. If your receiver accepts 3.3V input, then it can be connected
-    directly to the Pi.
-    """
     MIN_OUT = -1
     MAX_OUT = 1
-
-    def __init__(self, gpio, invert=False, jitter=0.025, no_action=None):
-        """
-        :param gpio: gpio pin connected to RC channel
-        :param invert: invert value of run() within [MIN_OUT,MAX_OUT]
-        :param jitter: threshold below which no signal is reported
-        :param no_action: value within [MIN_OUT,MAX_OUT] if no RC signal is
-                          sent. This is usually zero for throttle and steering
-                          being the center values when the controls are not
-                          pressed.
-        """
+    def __init__(self, cfg, debug=False):
         import pigpio
-        self.pi = pigpio.pi()
-        self.gpio = gpio
-        self.high_tick = None
-        self.period = None
-        self.high = None
+
+        self.rc1 = pigpio.pi()
+        self.rc2 = pigpio.pi()
+        self.rc3 = pigpio.pi()
+
+        # standard variables
+        self.high_tick1 = None
+        self.high_tick2 = None
+        self.high_tick3 = None
+        self.high1 = None
+        self.high2 = None
+        self.high3 = None
```

Comment:
Instead of having triplication of all variables here, I suggest to use a list of len 3, that make the whole code more readable and shorter, because you'll just have to loop/iterate over these further down.

Reply:
This seems like just a style preference thing.  looping over lists just isn't my preferred way to do it for only three variables. I find it harder to read.